app/assets/javascripts/discourse/app/components/composer/toolbar-buttons.gjs

```diff
@@ -0,0 +1,68 @@
+import Component from "@glimmer/component";
+import { fn, hash } from "@ember/helper";
+import { action } from "@ember/object";
+import { eq } from "truth-helpers";
+import DButton from "discourse/components/d-button";
+import concatClass from "discourse/helpers/concat-class";
+import icon from "discourse/helpers/d-icon";
+import ToolbarPopupMenuOptions from "select-kit/components/toolbar-popup-menu-options";
+
+export default class ComposerToolbarButtons extends Component {
+  firstButton = this.args.data.groups.mapBy("buttons").flat().firstObject;
+
+  @action
+  applyCondition(button) {
+    return button.condition === undefined
+      ? true
+      : button.condition(this.args.data);
+  }
+
+  @action
+  tabIndex(button) {
+    return button === this.firstButton ? 0 : button.tabindex;
+  }
+
+  <template>
+    {{#each @data.groups as |group|}}
```

Comment:
We should add `key="..."` to prevent the `each` from recreating the elements on every redraw